src/main/java/oscar/command/DeadlineCommand.java

```diff
@@ -0,0 +1,80 @@
+package oscar.command;
+
+import java.time.LocalDateTime;
+import java.time.format.DateTimeParseException;
+
+import oscar.essential.Storage;
+import oscar.essential.TaskList;
+import oscar.exception.OscarException;
+import oscar.task.Deadline;
+import oscar.task.Task;
+
+/**
+ * Command to create a new deadline task.
+ */
+public class DeadlineCommand extends Command {
+    private final String details;
+
+    /**
+     * Instantiates a deadline command.
+     *
+     * @param details Description of deadline task.
+     */
+    public DeadlineCommand(String details) {
+        super(false);
+        this.details = details;
+    }
+
+    /**
+     * Creates a new deadline task and save it to the task list.
+     *
+     * @param tasks ArrayList of tasks.
+     * @param storage File loading and saving handler.
+     * @throws OscarException Incorrect format of deadline command.
+     */
+    @Override
+    public void execute(TaskList tasks, Storage storage) throws OscarException {
+        String[] validatedDetails = validate();
```

Comment:
Great naming here! Abided the convention of plural form to be used on names representing a collection of objects!